cylc/flow/config.py

```diff
@@ -1521,55 +1523,37 @@ def run_mode(self, *reqmodes):
         else:
             return mode
 
-    def check_tasks(self):
-        """Call after all tasks are defined.
-
-        ONLY IF VALIDATING THE SUITE
-        because checking conditional triggers below may be slow for
-        huge suites (several thousand tasks).
-        Note:
-          (a) self.cfg['runtime'][name]
-              contains the task definition sections of the flow.cylc file.
-          (b) self.taskdefs[name]
-              contains tasks that will be used, defined by the graph.
-        Tasks (a) may be defined but not used (e.g. commented out of the
-        graph)
-        Tasks (b) may not be defined in (a), in which case they are dummied
-        out.
-        """
+    def _check_task_event_handlers(self):
+        """Check custom event handler templates can be expanded.
 
+        Ensures that any %(template_variables)s in task event handlers
+        are present in the data that will be passed to them when called
+        (otherwise they will fail).
+        """
         for taskdef in self.taskdefs.values():
-            taskdef.check_for_explicit_cycling()
-            # Check custom event handler templates compat with task meta
             if taskdef.rtconfig['events']:
-                subs = dict((key, key) for key in self.TASK_EVENT_TMPL_KEYS)
-                for key, value in self.cfg['meta'].items():
-                    subs['suite_' + key.lower()] = value
-                subs.update(taskdef.rtconfig['meta'])
-                # Back compat.
-                try:
-                    subs['task_url'] = subs['URL']
```

Comment:
This is the back-compat for `task_url`, however the back-compat for `suite_url` seems to be missing from here, though present in `task_event_mgr`?

Centralised this logic to remove the problem.